third-party/chpl-venv/venv-use-sys-python.py

```diff
@@ -4,32 +4,97 @@
+
+    root is the abs path to $(chpl_home)/third-party/chpl-venv/$(platform)/$(python_version)
+    """
+
+    if os.path.isdir(root) and not os.path.islink(root):
+        for f in os.listdir(root):
+            rw = False
+
+            abs_f = os.path.join(root, f)
+            # abs_f is the abs path to some file system object under root
+            if os.path.isfile(abs_f) and not os.path.islink(abs_f):
+                # abs_f is a file
+                with open(abs_f) as fin:
+                    # read file abs_f and examine the first line
+                    lines = fin.readlines()
+                    if len(lines) > 0:
+                        line = lines[0]
+                        if re.match(r'#!/.*/bin/python', line):
+                            # first line is a shebang that points to python
+                            if not re.match(r'#!/usr/bin/python', line):
+                                # shebang points to some non-standard location, so generate a new shebang
+                                # the new shebang should be #!/usr/bin/python, #!/usr/bin/python2, etc
+                                newline = re.sub(r'^.*/bin/python', '#!/usr/bin/python', line)
+                                lines[0] = newline
+                                rw = True
+
+                if rw:
+                    # rewrite the file
+                    print('Rewriting shebang for {0}'.format(abs_f))
+                    print(' - from\t{0}'.format(line.rstrip()))
+                    print(' -   to\t{0}'.format(newline.rstrip()))
+
+                    with open (abs_f, 'w') as fout:
+                        for line in lines:
+                            fout.write(line)
+
+    else:
+        print('error, rw_shebangs(): {0} not found'.format(root))
+        sys.exit(2)
 
-# Remove the virtualenv pythons as well as references to them in
-# scripts. This effectively forces the virtualenv scripts to use the
-# system python.
-def remove_venv_python_references(venv_bin_dir):
-    for f in os.listdir(venv_bin_dir):
-        abs_f = os.path.join(venv_bin_dir, f)
-        if f.startswith("python"):
-            print('Removing {0}'.format(abs_f))
-            os.remove(abs_f)
-        else:
-            with open(abs_f) as fin:
-                lines = fin.readlines()
-
-            python_shebang = '#!/usr/bin/env python'
-            if len(lines) > 0  and lines[0].startswith (python_shebang):
-                print('Updating shebang for {0}'.format(abs_f))
-                lines[0] = '{0}\n'.format(python_shebang)
-
-            with open (abs_f, 'w') as fout:
-               for line in lines:
-                   fout.write(line)
 
 def main():
-    venv_bin_dir = sys.argv[1]
-    remove_venv_python_references(venv_bin_dir)
+    chpl_venv_install_dir = sys.argv[1]
+    rm_python_bins(os.path.join(chpl_venv_install_dir, 'chpl-virtualenv', 'bin'))
+    rw_shebangs(chpl_venv_install_dir)
 
 
 if __name__ == '__main__':
```

Comment:
maybe do a check for `len(sys.argv)` here and pass `sys.argv[1]` to `main()` as an argument (`chpl_venv_install_dir`).